src/box/xrow_io.cc

```diff
@@ -102,3 +105,54 @@ coio_write_xrow(struct iostream *io, const struct xrow_header *row)
 		diag_raise();
 }
 
+uint64_t xrow_stream_flush_size = 16384;
+TWEAK_UINT(xrow_stream_flush_size);
+
+void
+xrow_stream_write(struct xrow_stream *stream, const struct xrow_header *row)
+{
+	/* Fixheader - length of packet. */
+	size_t fixheader_len = mp_sizeof_uint(UINT32_MAX);
+	size_t len = 0;
+	len += xrow_header_sizeof(row, row->sync);
+	for (int i = 0; i < row->bodycnt; i++) {
+		len += row->body[i].iov_len;
+	}
+	char *data = (char *)xlsregion_alloc(&stream->lsregion,
+					     len + fixheader_len,
+					     ++stream->lsr_id);
+	*(data) = 0xce; /* MP_UINT32 */
+	store_u32(data + 1, mp_bswap_u32(len));
+	data = data + fixheader_len;
+	data += xrow_header_encode(row, row->sync, data);
```

Comment:
Could you also make `xlsregion_reserve(XROW_HEADER_LEN_MAX)`, do `xrow_header_encode()`, and then `xlsregion_alloc(actually_encoded_size)`? Then you wouldn't need to do the relatively expensive size calculation before the encoding + wouldn't have gaps in the lsregion.

Reply:
Yes, indeed. It didn't come to my mind. Done, thanks!